### ISSUE TITLE
Lancero kernel check

### ISSUE THREAD

**Issue by joefowler** [usnistgov/quantum-sensors-drivers#3]
**[Original report](https://bitbucket.org/nist_microcal/nist_qsp_drivers/issues/3) by Dark Tyr (Bitbucket: [darktyr](https://bitbucket.org/darktyr), GitHub: [darktyr](https://github.com/darktyr)).**

----------------------------------------

I think the lancero driver is failing the kernel check again. 

terminal:
make
make -C /lib/modules/4.4.0-57-generic/build M=/home/darktyr/gitrepo/nist_qsp_drivers/lancero/Lancero-RELEASE/C/driver modules
make[1]: Entering directory '/usr/src/linux-headers-4.4.0-57-generic'
  CC [M]  /home/darktyr/gitrepo/nist_qsp_drivers/lancero/Lancero-RELEASE/C/driver/lancero-base.o
/home/darktyr/gitrepo/nist_qsp_drivers/lancero/Lancero-RELEASE/C/driver/lancero-base.c: In function ‘lancero_init’:
/home/darktyr/gitrepo/nist_qsp_drivers/lancero/Lancero-RELEASE/C/driver/lancero-base.c:3022:38: error: macro "__DATE__" might prevent reproducible builds [-Werror=date-time]
  printk(KERN_INFO DRV_NAME " built " __DATE__ " " __TIME__ "\n");
                                      ^
/home/darktyr/gitrepo/nist_qsp_drivers/lancero/Lancero-RELEASE/C/driver/lancero-base.c:3022:51: error: macro "__TIME__" might prevent reproducible builds [-Werror=date-time]
  printk(KERN_INFO DRV_NAME " built " __DATE__ " " __TIME__ "\n");
                                                   ^
cc1: some warnings being treated as errors
scripts/Makefile.build:258: recipe for target '/home/darktyr/gitrepo/nist_qsp_drivers/lancero/Lancero-RELEASE/C/driver/lancero-base.o' failed
make[2]: *** [/home/darktyr/gitrepo/nist_qsp_drivers/lancero/Lancero-RELEASE/C/driver/lancero-base.o] Error 1
Makefile:1420: recipe for target '_module_/home/darktyr/gitrepo/nist_qsp_drivers/lancero/Lancero-RELEASE/C/driver' failed
make[1]: *** [_module_/home/darktyr/gitrepo/nist_qsp_drivers/lancero/Lancero-RELEASE/C/driver] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-4.4.0-57-generic'
Makefile:22: recipe for target 'lancero.ko' failed
make: *** [lancero.ko] Error 2

uname -a output:
Linux DarkNight-i5-linux 4.4.0-57-generic #78-Ubuntu SMP Fri Dec 9 23:50:32 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

There are warnings being treated as errors here. If we can get the flag `-Wno-error=date-time` into the Make system, then I think it would be solved. Possible patch:


```
#!c

  lancero-src := lancero-base.c lancero-user.c
  ccflags-y := -Wno-error=date-time
```

The first line is already there. There's some evidence that adding the second line would do it. Can you try at home?

**Comment by joefowler** [comment]
**Original comment by Dark Tyr (Bitbucket: [darktyr](https://bitbucket.org/darktyr), GitHub: [darktyr](https://github.com/darktyr)).**

----------------------------------------

When the cflags line is inserted under lancero-src, it still fails. It does however work when placed at the top of the file under DRIVER=lancero.ko

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

@oneilg reported the same problem under Ubuntu 16.04 LTS. (I assume that's what John was using, too.) Anyway, Galen is in kernel 4.4.0-75. I am using John's fix and will commit it soon.

**Comment by joefowler** [comment]
**Original comment by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

Change lancero Makefile so a "fatal warning" is treated as non-fatal.

This is needed on Ubuntu 16.04 (kernel 4.4.0). Fixes #3.
